src/main/java/me/itzg/helpers/modrinth/ProjectRef.java

```diff
@@ -36,32 +42,48 @@ public class ProjectRef {
     private final String versionNumber;
 
     public static ProjectRef parse(String projectRef) {
-        final Matcher m = PROJECT_REF.matcher(projectRef);
-        if (!m.matches()) {
-            throw new InvalidParameterException("Invalid project reference: " + projectRef);
+        // First, try to split into potential loader prefix and the rest
+        final int firstColon = projectRef.indexOf(':');
+        Loader loaderRef = null;
+        String rest = projectRef;
+        
+        if (firstColon > 0) {
+            final String prefix = projectRef.substring(0, firstColon);
+            if (VALID_LOADERS.contains(prefix.toLowerCase())) {
+                loaderRef = Loader.valueOf(prefix);
```

Comment:
Probably should be this since the enum names are all lowercase too

```suggestion
                loaderRef = Loader.valueOf(prefix.toLowerCase());
```